nixos/modules/services/mail/public-inbox.nix

```diff
@@ -0,0 +1,569 @@
+          `services.public-inbox.settings.publicinboxwatch.spamcheck' to null.
+        '';
+      }
+      { assertion = cfg.path != [] || !useSpamAssassin;
+        message = ''
+          public-inbox is configured to use SpamAssassin, but there is
+          no spamc executable in services.public-inbox.path.  If you
+          don't need spam checking, set
+          `services.public-inbox.settings.publicinboxmda.spamcheck' and
+          `services.public-inbox.settings.publicinboxwatch.spamcheck' to null.
+        '';
+      }
+    ];
+    services.public-inbox.settings =
+      filterAttrsRecursive (n: v: v != null) {
+      publicinbox = mapAttrs (n: filterAttrs (n: v: n != "description")) cfg.inboxes;
+    };
+    users = {
+      users.public-inbox = {
+        # Use runCommand instead of linkFarm,
+        # because Postfix rejects .forward if it's a symlink.
+        home = pkgs.runCommand "public-inbox-home" {} (''
+          install -D -p ${environment.PI_CONFIG} $out/.public-inbox/config
+          ln -s ${stateDir}/emergency $out/.public-inbox/emergency
+          ln -s ${stateDir}/spamassassin $out/.spamassassin
+        '' + optionalString cfg.mda.enable ''
+          cp ${let env = concatStringsSep " " (mapAttrsToList (n: v: "${n}=${escapeShellArg v}") environment); in
+            pkgs.writeText "forward" ''
+              |"env ${env} PATH=\"${makeBinPath cfg.path}:$PATH\" ${cfg.package}/bin/public-inbox-mda ${escapeShellArgs cfg.mda.args}
+          ''} $out/.forward
+        '');
+        group = "public-inbox";
+        isSystemUser = true;
+      };
+      groups.public-inbox = {};
+    };
+    networking.firewall = mkIf cfg.openFirewall
+      { allowedTCPPorts = mkMerge [
+          (mkIf (cfg.http.enable && types.port.check cfg.http.port) [ cfg.http.port ])
+          (mkIf (cfg.imap.enable && types.port.check cfg.imap.port) [ cfg.imap.port ])
+          (mkIf (cfg.nntp.enable && types.port.check cfg.nntp.port) [ cfg.nntp.port ])
+        ];
+      };
+    systemd.sockets = mkMerge (map (proto:
+      mkIf (cfg.${proto}.enable && cfg.${proto}.port != null)
+        { "public-inbox-${proto}d" = {
+            listenStreams = [ (toString cfg.${proto}.port) ];
+            wantedBy = [ "sockets.target" ];
+          };
+        }
+      ) [ "http" "imap" "nntp" ]);
+    systemd.services = mkMerge [
+      (mkIf cfg.http.enable
+        { public-inbox-httpd = {
+          inherit environment;
+          after = [ "public-inbox-init.service" "public-inbox-watch.service" ];
+          requires = [ "public-inbox-init.service" ];
+          serviceConfig = serviceConfig "httpd" // {
+            ExecStart = escapeShellArgs (
+              [ "${cfg.package}/bin/public-inbox-httpd" ] ++
+              cfg.http.args ++
+              [ (pkgs.writeText "public-inbox.psgi" ''
+                #!${cfg.package.fullperl} -w
+                use strict;
```

Comment:
This file comes from public-inbox, and had a license header attached (which was still present in my PR).   I don't think we can just remove it.

Reply:
I would re-add it if you insist but here is why I removed it:
1. I see this as being a configuration to use `public-inbox`, not a derived work based upon `public-inbox`.
2. It hardly meets a sane [threshold of originality](https://en.wikipedia.org/wiki/Threshold_of_originality) as it's a textbook application of [Plack::Builder](https://metacpan.org/pod/Plack::Builder).
3. The code is not actually the same due to nix macros, hence the copyright would not be accurate.